### PR TITLE
signup screen

### PR DESCRIPTION
**Pull request by adedayo142**
This PR was created to build the signup screen

### CODE REVIEW

File: lib/ui/utilities/validation.dart [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import 'package:receipe_app/ui/utilities/string_utils.dart';
 
 class Validation {
-
   static String? validateEmail(String? s) {
     if (StringUtil.isEmpty(s)) {
       return "Email cannot be empty";
```
Comment (moak13) [inline lib/ui/utilities/validation.dart:6]: kindly help with this as part of the internalization

File: lib/ui/utilities/validation.dart [hunk 2]
```diff
@@ -12,12 +11,11 @@ class Validation {
     }
   }
 
-    static String? validateField(String? s, {String? errorMessage}) {
+  static String? validateField(String? s, {String? errorMessage}) {
     if (StringUtil.isEmpty(s)) {
       return errorMessage ?? "Required";
```
Comment (moak13) [inline lib/ui/utilities/validation.dart:16]: kindly help with the required as well

File: lib/ui/views/login/login_view.dart [hunk 3]
```diff
@@ -58,7 +60,8 @@ class LoginView extends StackedView<LoginViewModel> with $LoginView {
                 alignment: Alignment.centerLeft,
                 child: Text(
                   "Please sign in to continue",
```
Comment (moak13) [inline lib/ui/views/login/login_view.dart:62]: kindly help with this

File: lib/ui/views/login/login_view.dart [hunk 4]
```diff
@@ -48,7 +49,8 @@ class LoginView extends StackedView<LoginViewModel> with $LoginView {
                 alignment: Alignment.centerLeft,
                 child: Text(
                   "Welcome",
```
Comment (moak13) [inline lib/ui/views/login/login_view.dart:51]: kindly help with this

File: lib/ui/views/signup/signup_view.dart [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                AppImages.logo,
+                height: 72.h,
+              ),
+              SizedBox(
+                height: 33.h,
+              ),
+              Align(
+                alignment: Alignment.centerLeft,
+                child: Text(
+                  "Sign up",
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:55]: kindly handle this

File: lib/ui/views/signup/signup_view.dart [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                      ?.copyWith(color: palette?.gray11),
+                ),
+              ),
+              SizedBox(
+                height: 4.h,
+              ),
+              Align(
+                alignment: Alignment.centerLeft,
+                child: Text(
+                  "Kindly fill the form to create a Kally Dish account",
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:66]: kindly handle this as well

File: lib/ui/views/signup/signup_view.dart [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+              SizedBox(
+                height: 24.h,
+              ),
+              TextFormField(
+                controller: firstNameController,
+                focusNode: firstNameFocusNode,
+                validator: Validation.validateField,
+                textInputAction: TextInputAction.next,
+                decoration: const InputDecoration(
+                    labelText: "First Name", hintText: "First Name"),
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:80]: kindly handle the texts here, `hint` should give and hint not the same text as the label.

File: lib/ui/views/signup/signup_view.dart [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+              ),
+              TextFormField(
+                controller: emailController,
+                focusNode: emailFocusNode,
+                autofillHints: const [AutofillHints.email],
+                keyboardType: TextInputType.emailAddress,
+                textInputAction: TextInputAction.next,
+                validator: Validation.validateEmail,
+                decoration: const InputDecoration(
+                    labelText: "Email Address",
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:104]: same as above

File: lib/ui/views/signup/signup_view.dart [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+              TextFormField(
+                controller: emailController,
+                focusNode: emailFocusNode,
+                autofillHints: const [AutofillHints.email],
+                keyboardType: TextInputType.emailAddress,
+                textInputAction: TextInputAction.next,
+                validator: Validation.validateEmail,
+                decoration: const InputDecoration(
+                    labelText: "Email Address", 
+                    hintText: "Email Address"),
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:105]: same as above

File: lib/ui/views/signup/signup_view.dart [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+              ),
+              TextFormField(
+                controller: passwordController,
+                focusNode: passwordFocusNode,
+                obscureText: viewModel.hidePassword,
+                  textInputAction: TextInputAction.done,
+                validator: Validation.validateField,
+                decoration: InputDecoration(
+                  labelText: "Password",
+                  hintText: "Password",
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:118]: handle this

File: lib/ui/views/signup/signup_view.dart [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+                          : Icons.visibility_off,
+                    ),
+                  ),
+                ),
+              ),
+              SizedBox(
+                height: 129.h,
+              ),
+              PrimaryButton(
+                buttonText: "Sign up",
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:133]: handle this

File: lib/ui/views/signup/signup_view.dart [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+                onTap: () {
+                  if (_signupKey.currentState?.validate() ?? false) {
+                    viewModel.signup();
+                  }
+                },
+              ),
+              SizedBox(
+                height: 16.h,
+              ),
+              Text.rich(TextSpan(
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:143]: kindly handle all the plain text here.

File: lib/ui/views/signup/signup_view.dart [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+   @override
+  void onViewModelReady(SignupViewModel viewModel) {
+    syncFormWithViewModel(viewModel);
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:167]: where is ur `super` call here?

**Review by moak13 — CHANGES_REQUESTED** [review]
Nice work, kindly take a look at my concerns below.

File: lib/ui/views/login/login_view.dart (hunk 3 above)
Comment (adedayo142) [reply lib/ui/views/login/login_view.dart:62]: Help in what way

File: lib/ui/utilities/validation.dart (hunk 2 above)
Comment (adedayo142) [reply lib/ui/utilities/validation.dart:16]: I should remove the question mark?

File: lib/ui/utilities/validation.dart (hunk 1 above)
Comment (adedayo142) [reply lib/ui/utilities/validation.dart:6]: I do not understand

File: lib/ui/views/signup/signup_view.dart (hunk 5 above)
Comment (adedayo142) [reply lib/ui/views/signup/signup_view.dart:55]: How

File: lib/ui/views/signup/signup_view.dart (hunk 13 above)
Comment (adedayo142) [reply lib/ui/views/signup/signup_view.dart:167]: it is with the dispose now

**Review by ikennaumeh — APPROVED** [review]
LGTM

File: lib/ui/views/signup/signup_view.dart (hunk 13 above)
Comment (moak13) [reply lib/ui/views/signup/signup_view.dart:172]: nope, its a separate logic on its own, so its meant to have it's super called as well.

File: lib/ui/utilities/validation.dart (hunk 2 above)
Comment (moak13) [reply lib/ui/utilities/validation.dart:17]: nope, add the required to the intl as well

File: lib/ui/views/login/login_view.dart [hunk 14]
```diff
@@ -84,19 +88,21 @@ class LoginView extends StackedView<LoginViewModel> with $LoginView {
                 obscureText: viewModel.hideText,
                 validator: Validation.validateField,
                 decoration: InputDecoration(
-                  labelText: "Password",
-                  hintText: "Password",
+                  labelText: S.current.password,
+                  hintText: S.current.password,
```
Comment (moak13) [inline lib/ui/views/login/login_view.dart:92]: this is meant to carry another string, eg: `please enter your password`

File: lib/ui/views/signup/signup_view.dart (hunk 6 above)
Comment (moak13) [reply lib/ui/views/signup/signup_view.dart:67]: you missed this

File: lib/ui/views/signup/signup_view.dart [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+                height: 24.h,
+              ),
+              TextFormField(
+                controller: firstNameController,
+                focusNode: firstNameFocusNode,
+                validator: Validation.validateField,
+                textInputAction: TextInputAction.next,
+                decoration:  InputDecoration(
+                    labelText: S.current.first_name, 
+                    hintText: S.current.first_name,
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:82]: this is meant to carry another string

File: lib/ui/views/signup/signup_view.dart [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+                height: 16.h,
+              ),
+              TextFormField(
+                controller: lastNameController,
+                focusNode: lastNameFocusNode,
+                validator: Validation.validateField,
+                textInputAction: TextInputAction.next,
+                decoration:  InputDecoration(
+                    labelText: S.current.last_name, 
+                    hintText: S.current.last_name,
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:95]: this is meant to carry another string

File: lib/ui/views/signup/signup_view.dart [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+              TextFormField(
+                controller: emailController,
+                focusNode: emailFocusNode,
+                autofillHints: const [AutofillHints.email],
+                keyboardType: TextInputType.emailAddress,
+                textInputAction: TextInputAction.next,
+                validator: Validation.validateEmail,
+                decoration:  InputDecoration(
+                    labelText: S.current.email_address, 
+                    hintText: S.current.email_address),
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:110]: this is meant to carry another string

File: lib/ui/views/signup/signup_view.dart [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+              ),
+              TextFormField(
+                controller: passwordController,
+                focusNode: passwordFocusNode,
+                obscureText: viewModel.hidePassword,
+                  textInputAction: TextInputAction.done,
+                validator: Validation.validateField,
+                decoration: InputDecoration(
+                  labelText: S.current.password,
+                  hintText: S.current.password,
```
Comment (moak13) [inline lib/ui/views/signup/signup_view.dart:123]: this is meant to carry another

File: lib/ui/utilities/validation.dart (hunk 2 above)
Comment (moak13) [reply lib/ui/utilities/validation.dart:17]: this is still hanging...

File: lib/ui/views/signup/signup_view.dart (hunk 6 above)
Comment (moak13) [reply lib/ui/views/signup/signup_view.dart:67]: I am not sure if this has been handled

File: lib/ui/views/signup/signup_view.dart (hunk 13 above)
Comment (moak13) [reply lib/ui/views/signup/signup_view.dart:172]: this is still missing its `super` call.

**Review by moak13 — CHANGES_REQUESTED** [review]
Kindly attend to this

File: lib/ui/utilities/validation.dart (hunk 2 above)
Comment (adedayo142) [reply lib/ui/utilities/validation.dart:17]: okay

File: lib/ui/views/signup/signup_view.dart (hunk 13 above)
Comment (adedayo142) [reply lib/ui/views/signup/signup_view.dart:172]: nothing like that in the login screen

**Review by moak13 — APPROVED** [review]

**Comment by moak13** [comment]
LGTM

**Pull request merged**